internal/storage/bucket_handle.go

```diff
@@ -512,13 +512,8 @@ func (b *bucketHandle) GetFolder(ctx context.Context, folderName string) (*gcs.F
 		return nil, err
 	}
 
-	folder := gcs.Folder{
-		Name:           clientFolder.Name,
-		Metageneration: clientFolder.Metageneration,
-		UpdateTime:     clientFolder.GetUpdateTime().AsTime(),
-	}
-
-	return &folder, err
+	folder := gcs.ConvertToGCSFolderFromGetFolderResponse(b.bucketName, clientFolder)
```

Comment:
nit = Can clientFolder be renamed to folderResponse or something to signify its a network response. There are too many similar names which hampers readability.